src/components/Disclosure/DisclosureContext.tsx

```diff
@@ -21,12 +24,22 @@ export const DisclosureToggleContext = React.createContext<
 >(undefined);
 
 export function DisclosureProvider(props: DisclosureProviderProps) {
-    const {size, disabled, defaultExpanded, arrowPosition, summary, keepMounted, onChange} = props;
+    const {
+        size,
+        disabled,
+        defaultExpanded,
+        arrowPosition,
+        summary,
+        keepMounted,
+        onChange,
+        expanded: controlledExpanded,
+    } = props;
     const [expanded, setExpanded] = React.useState(() => Boolean(defaultExpanded));
+    const controlledMode = controlledExpanded !== undefined;
 
     const handleToggle = (event: React.SyntheticEvent) => {
         setExpanded((prev) => !prev);
-        onChange(event, !expanded);
+        onChange({event, expanded: controlledMode ? undefined : !expanded});
```

Comment:
1) Why in controlled mode `expanded` is undefined? It should always be defined
2) Object argument is not convenient in use, I would leave 2 arguments but swap them. Then basic usage will be like this:
```jsx
const [expanded, setExpanded] = React.useState(true);

<Disclosure expanded={expanded} onChange={setExpanded}/>
```

What do you think?

Reply:
in `onChange` callback new value of `expanded` should be passed. But in controlled mode we don't know this new value. And that's why I passed `undefined` to it.

Reply:
In controlled mode we pass next value. That’s how in works in form controls

Reply:
But how can we find out the next value?..

Reply:
It is the opposite of current value, we pass the value we want to be next in onChange handle